src/base/utils/misc.cpp

```diff
@@ -625,8 +627,16 @@ void Utils::Misc::openFolderSelect(const QString &absolutePath)
     if ((output == "dolphin.desktop") || (output == "org.kde.dolphin.desktop"))
         proc.startDetached("dolphin", QStringList() << "--select" << Utils::Fs::toNativePath(path));
     else if ((output == "nautilus.desktop") || (output == "org.gnome.Nautilus.desktop")
-             || (output == "nautilus-folder-handler.desktop"))
-        proc.startDetached("nautilus", QStringList() << "--no-desktop" << Utils::Fs::toNativePath(path));
+                 || (output == "nautilus-folder-handler.desktop")) {
+        proc.start("nautilus", {"--version"});
+        proc.waitForFinished();
+        const QString nautilusVerStr = QString(proc.readLine().simplified()).remove(QRegExp("[^0-9.]"));
+        using NautilusVersion = Utils::Version<int, 3>;
+        if (NautilusVersion::tryParse(nautilusVerStr, {1, 0, 0}) > NautilusVersion {3, 28})
+            proc.startDetached("nautilus", QStringList() << Utils::Fs::toNativePath(path));
+        else
+            proc.startDetached("nautilus", QStringList() << "--no-desktop" << Utils::Fs::toNativePath(path));
```

Comment:
https://github.com/qbittorrent/qBittorrent/pull/8929#discussion_r188224554

Reply:
I'm not sure that code refactoring is appropriate for a particular issue.
Is it necessary? if so, change to curly brackets

Reply:
>Is it necessary? 

We use the new form (curly brackets) everywhere when writing new code, since you are adding new code & refactoring old code, I would recommend you change it.

Reply:
Ok.

``pythonProc.start("python3", QStringList() << "--version", QIODevice::ReadOnly);``
to
``pythonProc.start("python3", {"--version"}, QIODevice::ReadOnly);``